be/src/runtime/sender_queue.cpp

```diff
@@ -786,13 +786,13 @@ Status DataStreamRecvr::PipelineSenderQueue::add_chunks(const PTransmitChunkPara
                 // We cannot early-return for short circuit, it may occur for parts of parallelism,
                 // and the other parallelism may need to proceed.
             }
-            if (_is_cancelled) {
-                clean_buffer_queues();
-                return Status::OK();
-            }
             _recvr->_num_buffered_bytes += chunk_bytes;
             COUNTER_ADD(metrics.peak_buffer_mem_bytes, chunk_bytes);
         }
+        //
+        if (_is_cancelled) {
+            clean_buffer_queues();
+        }
```

Comment:
should move them outside the outer `}`, as the `keep_order` branch should also consider this case.